From: Carlos Garnacho <carlosg@gnome.org>
Date: Wed, 7 Apr 2021 11:27:56 +0200
Subject: docs: Change lower datetime bound in doc/tests about limits

Although SQLite supports dates starting from year 0000-01-01 (1 BC
in ISO8601 parlance), we are porting to using GDateTime internally
which changes the lower boundary to 0001-01-01.

This date range change is unlikely to have real life consequences,
and OTOH using GDateTime allows us to drop our own ISO8601 parsing
and architecture dependent code, which has larger consequences with
eg. 32-bit time_t.

Origin: upstream, commit:3028e54d7432accdfcb43bc79f52fcd474e9eb82
Applied-Upstream: 3.2.0~alpha
---
 docs/reference/libtracker-sparql/limits.xml | 2 +-
 tests/libtracker-data/datetime/direct-1.out | 2 +-
 tests/libtracker-data/datetime/direct-1.rq  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/docs/reference/libtracker-sparql/limits.xml b/docs/reference/libtracker-sparql/limits.xml
index 6119fe9..4e269cf 100644
--- a/docs/reference/libtracker-sparql/limits.xml
+++ b/docs/reference/libtracker-sparql/limits.xml
@@ -126,7 +126,7 @@
     <para>
       Integers are 64 bit wide. Floating point numbers have IEEE764
       double precision. Dates/times have microsecond precision, and may
-      range between 0000-01-01 00:00:00 and 9999-12-31 23:59:59.
+      range between 0001-01-01 00:00:00 and 9999-12-31 23:59:59.
     </para>
   </sect2>
 </part>
diff --git a/tests/libtracker-data/datetime/direct-1.out b/tests/libtracker-data/datetime/direct-1.out
index 566938b..a26f106 100644
--- a/tests/libtracker-data/datetime/direct-1.out
+++ b/tests/libtracker-data/datetime/direct-1.out
@@ -1 +1 @@
-"0000-01-01T00:00:00Z"	"2020-01-01T00:00:00Z"	"9999-01-01T00:00:00Z"
+"0001-01-01T00:00:00Z"	"2020-01-01T00:00:00Z"	"9999-01-01T00:00:00Z"
diff --git a/tests/libtracker-data/datetime/direct-1.rq b/tests/libtracker-data/datetime/direct-1.rq
index 51de510..a9f90e7 100644
--- a/tests/libtracker-data/datetime/direct-1.rq
+++ b/tests/libtracker-data/datetime/direct-1.rq
@@ -1 +1 @@
-SELECT ("0000-01-01T00:00:00Z"^^xsd:dateTime AS ?d1) ("2020-01-01T00:00:00Z"^^xsd:dateTime AS ?d2) ("9999-01-01T00:00:00Z"^^xsd:dateTime AS ?d3) {}
\ No newline at end of file
+SELECT ("0001-01-01T00:00:00Z"^^xsd:dateTime AS ?d1) ("2020-01-01T00:00:00Z"^^xsd:dateTime AS ?d2) ("9999-01-01T00:00:00Z"^^xsd:dateTime AS ?d3) {}
\ No newline at end of file
